pyro/infer/mcmc/api.py

```diff
@@ -313,6 +313,9 @@ def __init__(self, kernel, num_samples, warmup_steps=None, initial_params=None,
                     if v.shape[0] != num_chains:
                         raise ValueError("The leading dimension of tensors in `initial_params` "
                                          "must match the number of chains.")
+                # FIXME: probably we want to use "spawn" method by default to avoid the error
+                # CUDA initialization error https://github.com/pytorch/pytorch/issues/2517
+                # even that we run MCMC in CPU.
```

Comment:
This error won't be detected in Travis or Mac. This happens when we use Linux (with default mp_context is "fork"), install PyTorch with GPU support, and run MCMC with multi-chains. The solution
> to work around, you need to make sure there are no CUDA calls prior to starting your subprocesses; torch.cuda.is_available() and torch.manual_seed both count as a CUDA calls
is hard to meet in Pyro (e.g. users usually run `pyro.set_rng_seed()` or run some PyTorch code involving checking if a tensor is CUDA tensor).

Reply:
Is "spawn" much slower as compared to "fork" or is the speed comparable? One of the reasons why we didn't have "spawn" is that it wasn't available in python 2, but since we have moved to python 3, we can make this change easily if there aren't any performance reasons blocking us. We can do it in a separate PR.

Reply:
I remember that fork is faster in Linux. I'll verify it again. If so, I'll ask PyTorch devs if they have a better solution for this issue.

Reply:
There is a better solution to this issue: that is to disable CUDA at the beginning of the notebook/script.
```
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
```
"spawn" method [does not play well with Jupyter notebook](https://github.com/pytorch/pytorch/issues/20375).